materials-data-facility/connect_server
# GitHub Action to Publish Lambda Docker Images

# Problem
We want to use docker images for our lambda functions

Solution to #84 

# Approach
Create a Dockerfile based on the AWS Python Lambda base image. The Dockerfile accepts a build arg `LAMBDA_SCRIPT` which is the name of the python module in the `aws` directory that actually contains the lambda function. This is baked into the docker image as the Entrypoint.

The GitHub action loops over the three lambda functions and creates a docker image for each one. Logs into AWS using OIDC link. Pushes images to ECR

# How to test
1. Review the actions on this PR to see the docker images being built
2. Test the docker image as follows:

```shell
% docker pull bengal1/submit_dataset:dockerize
% docker run --rm -it -p 9000:8080 bengal1/submit_dataset:dockerize
```
in a separate shell,
```shell
% curl "http://localhost:9000/2015-03-31/functions/function/invocations" -d '{}'
```

You should see an error from the lambda function itself showing that it attempted the access the auth information (which will not be present in this simple request)

```
{"errorMessage": "'requestContext'", "errorType": "KeyError", "requestId": "d8be889f-b2e5-4527-9973-183a8b4a009c", "stackTrace": ["  File \"/var/task/submit_dataset.py\", line 43, in lambda_handler\n    name = event['requestContext']['authorizer']['name']\n"]}
```
